### ISSUE TITLE
Face detection back camera model in Python

### ISSUE THREAD

**Issue by romaniukm** [google/mediapipe#2112]
The present Python solution seems to support the front camera model only. Are there any plans to add the back camera model in Python?

**Comment by sgowroji** [comment]
Hi @romaniukm, Could you please provide your code changes and steps to reproduce your error. Thanks!

**Comment by romaniukm** [comment]
Ho @sgowroji , I was just trying to run the Python face detection example: https://google.github.io/mediapipe/solutions/face_detection.html#python-solution-api

It looks like there is no way to switch to the back-facing camera model. By looking at the Python code https://github.com/google/mediapipe/blob/master/mediapipe/python/solutions/face_detection.py I figured out that the model is always loaded from `BINARYPB_FILE_PATH = 'mediapipe/modules/face_detection/face_detection_front_cpu.binarypb'` and the face_detection_back model is only available in tflite format so simply changing the path won't work.

**Comment by samlaf** [comment]
@romaniukm I think you're confusing protobuf (.pbtxt and the compiled versions .binarypb) formats and tensorflow lite (.tflite) formats.
Protobuf files describe the computation graph, whereas tflite is actually used inside of the InferenceCalculator node of mediapipe graphs.
![image](https://user-images.githubusercontent.com/9342524/121834982-0bcfd100-cc9e-11eb-9be5-2a9b7749e2ba.png)

So you can just create a .pbtxt yourself to run the back .tflite trained model (not sure why mediapipe hasn't provided this yet...)
```
# MediaPipe graph to detect faces. (CPU input, and inference is executed on
# CPU.)
#
# It is required that "face_detection_front.tflite" is available at
# "mediapipe/modules/face_detection/face_detection_front.tflite"
# path during execution.
#
# EXAMPLE:
#   node {
#     calculator: "FaceDetectionFrontCpu"
#     input_stream: "IMAGE:image"
#     output_stream: "DETECTIONS:face_detections"
#   }

type: "FaceDetectionBackCpu"

# CPU image. (ImageFrame)
input_stream: "IMAGE:image"

# Detected faces. (std::vector<Detection>)
# NOTE: there will not be an output packet in the DETECTIONS stream for this
# particular timestamp if none of faces detected. However, the MediaPipe
# framework will internally inform the downstream calculators of the absence of
# this packet so that they don't wait for it unnecessarily.
output_stream: "DETECTIONS:detections"

# Converts the input CPU image (ImageFrame) to the multi-backend image type
# (Image).
node: {
  calculator: "ToImageCalculator"
  input_stream: "IMAGE_CPU:image"
  output_stream: "IMAGE:multi_backend_image"
}

# Transforms the input image into a 256x256 tensor while keeping the aspect
# ratio (what is expected by the corresponding face detection model), resulting
# in potential letterboxing in the transformed image.
node: {
  calculator: "ImageToTensorCalculator"
  input_stream: "IMAGE:multi_backend_image"
  output_stream: "TENSORS:input_tensors"
  output_stream: "MATRIX:transform_matrix"
  options: {
    [mediapipe.ImageToTensorCalculatorOptions.ext] {
      output_tensor_width: 256
      output_tensor_height: 256
      keep_aspect_ratio: true
      output_tensor_float_range {
        min: -1.0
        max: 1.0
      }
      border_mode: BORDER_ZERO
    }
  }
}

# Runs a TensorFlow Lite model on CPU that takes an image tensor and outputs a
# vector of tensors representing, for instance, detection boxes/keypoints and
# scores.
node {
  calculator: "InferenceCalculator"
  input_stream: "TENSORS:input_tensors"
  output_stream: "TENSORS:detection_tensors"
  options: {
    [mediapipe.InferenceCalculatorOptions.ext] {
      model_path: "mediapipe/modules/face_detection/face_detection_back.tflite"
      delegate { xnnpack {} }
    }
  }
}

# Performs tensor post processing to generate face detections.
node {
  calculator: "FaceDetectionFrontCommon"
  input_stream: "TENSORS:detection_tensors"
  input_stream: "MATRIX:transform_matrix"
  output_stream: "DETECTIONS:detections"
}
```
Notice that the only difference with the front model is that we use `face_detection_back.tflite`, but also we need to change the `output_tensor_width` and `output_tensor_height` in `ImageToTensorCalculator` (since the back model is trained with larger images, probably to be able to detect faces that are farther).

**Comment by jiuqiant** [comment]
Thanks @samlaf.

Yes, you can swap the face detection model to achieve this. Here are the steps to achieve this:
1) Go to your local python package directory, such as /Users/your_name/mp_env/lib/python3.9/site-packages/mediapipe
2) Remove modules/face_detection/face_detection_front.tflite
3) Copy https://github.com/google/mediapipe/blob/master/mediapipe/modules/face_detection/face_detection_back.tflite to modules/face_detection and rename it to be face_detection_front.tflite.

This workaround above is hacky. But, I think it should just work. The official support is on its way.

**Comment by sgowroji** [comment]
Hi @romaniukm, Could you please respond that the above comments resolved your query. Thanks!

**Comment by romaniukm** [comment]
> Thanks @samlaf.
> 
> Yes, you can swap the face detection model to achieve this. Here are the steps to achieve this:
> 
> 1. Go to your local python package directory, such as /Users/your_name/mp_env/lib/python3.9/site-packages/mediapipe
> 2. Remove modules/face_detection/face_detection_front.tflite
> 3. Copy https://github.com/google/mediapipe/blob/master/mediapipe/modules/face_detection/face_detection_back.tflite to modules/face_detection and rename it to be face_detection_front.tflite.
> 
> This workaround above is hacky. But, I think it should just work. The official support is on its way.

This looks like the simpler solution but it didn't work for me:

```
  File "/home/[redacted]/lib/python3.8/site-packages/mediapipe/python/solutions/face_detection.py", line 103, in process
    return super().process(input_data={'image': image})
  File "/home/[redacted]/lib/python3.8/site-packages/mediapipe/python/solution_base.py", line 307, in process
    self._graph.add_packet_to_input_stream(
RuntimeError: Graph has errors: 
Calculator::Open() for node "inferencecalculator__InferenceCalculatorCpu" failed: ; Failed to load model from path /home/[redacted]/lib/python3.8/site-packages/mediapipe/modules/face_detection/face_detection_front.tflite
```

**Comment by romaniukm** [comment]
> Protobuf files describe the computation graph, whereas tflite is actually used inside of the InferenceCalculator node of mediapipe graphs.

This looks like it might be the missing pbtxt file, but how do I convert it to .binarypb?

**Comment by samlaf** [comment]
You can feed either the compiled .binarypb or the pbtxt file into a mediapipe graph, so you don't need to compile it.

**Comment by jiuqiant** [comment]
Released face detection model selection in v0.8.6. See https://github.com/google/mediapipe/blob/139237092fedef164a17840aceb4e628244f8173/mediapipe/python/solutions/face_detection.py#L80
